cms/static/js/views/utils/create_library_utils.js

```diff
@@ -1,58 +1,17 @@
 /**
  * Provides utilities for validating libraries during creation.
  */
-define(["jquery", "underscore", "gettext", "js/views/utils/view_utils"],
-    function ($, _, gettext, ViewUtils) {
+define(["jquery", "gettext", "js/views/utils/view_utils", "js/views/utils/create_utils_base"],
+    function ($, gettext, ViewUtils, CreateUtilsFactory) {
         "use strict";
         return function (selectors, classes) {
-            var validateTotalKeyLength, setNewLibraryFieldInErr, hasInvalidRequiredFields,
-                createLibrary, validateFilledFields, configureHandlers;
+            var keyLengthViolationMessage = gettext("The combined length of the organization and library code fields cannot be more than <%=limit%> characters.")
+            var keyFieldSelectors = [selectors.org, selectors.number];
+            var nonEmptyCheckFieldSelectors = [selectors.name, selectors.org, selectors.number];
 
-            var validateRequiredField = ViewUtils.validateRequiredField;
-            var validateURLItemEncoding = ViewUtils.validateURLItemEncoding;
+            CreateUtilsFactory.call(this, selectors, classes, keyLengthViolationMessage, keyFieldSelectors, nonEmptyCheckFieldSelectors);
```

Comment:
Thanks, I will look over that link. I do understand .call and this (I think!), but I'm still stumped by how the "return" works. To me, it feels like line 12 should be

return CreateUtilsFactory.call(this, selectors, classes, keyLengthViolationMessage, keyFieldSelectors, nonEmptyCheckFieldSelectors);

But there is also a further line of code beyond that, defining the create function.